.github/workflows/test.yml

```diff
@@ -1,8 +1,11 @@
 name: Test
 
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - master
+      - release
+  pull_request:
```

Comment:
I depend on tests running on push and pull_request both for two reasons:
- there is a flacky test, which fails 20% of the time, with two tests in parallel it makes the probability both fail 20% * 20% -> 4%
- sometimes I test things without opening a PR at all, just to see if they pass

Reply:
While the latter is a matter of preference and I am open to change, the former has practical implications for now.